File: backend/src/main/java/com/backend/domain/analysis/entity/Score.java
package com.backend.domain.analysis.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "score")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Score {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "analysis_result_id", nullable = false)
    private AnalysisResult analysisResult;

    @Column(nullable = false)
    private int readmeScore;

    @Column(nullable = false)
    private int testScore;

    @Column(nullable = false)
    private int commitScore;

    @Column(nullable = false)
    private int cicdScore;

    // 종합 점수
    @Transient
    public int getTotalScore() {
        return readmeScore + testScore + commitScore + cicdScore;
    }

    @Builder
    public Score(AnalysisResult analysisResult, int readmeScore,
                 int testScore, int commitScore, int cicdScore) {
        this.analysisResult = analysisResult;
        if (analysisResult != null) {
            analysisResult.assignScore(this);
        }

        this.readmeScore = readmeScore;
        this.testScore = testScore;
        this.commitScore = commitScore;
        this.cicdScore = cicdScore;
    }
}

File: backend/src/main/java/com/backend/domain/analysis/entity/Score.kt
package com.backend.domain.analysis.entity

import com.backend.domain.repository.entity.Repositories
import jakarta.persistence.*

@Entity
@Table(name = "score")
open class Score protected constructor(
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "analysis_result_id", nullable = false)
    open var analysisResult: AnalysisResult,

    @Column(nullable = false)
    open var readmeScore: Int,

    @Column(nullable = false)
    open var testScore: Int,

    @Column(nullable = false)
    open var commitScore: Int,

    @Column(nullable = false)
    open var cicdScore: Int,
) {
    protected constructor() : this(
        analysisResult = AnalysisResult.create(
            repositories = Repositories.create(
                user = com.backend.domain.user.entity.User(),
                name = "",
                description = null,
                htmlUrl = "",
                publicRepository = false,
                mainBranch = "main"
            ),
            summary = "",
            strengths = "",
            improvements = ""
        ),
        readmeScore = 0,
        testScore = 0,
        commitScore = 0,
        cicdScore = 0
    )

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @get:Transient
    open val totalScore: Int
        get() = readmeScore + testScore + commitScore + cicdScore

    init {
        analysisResult.assignScore(this)
    }

    companion object {
        @JvmStatic
        fun create(
            analysisResult: AnalysisResult,
            readmeScore: Int,
            testScore: Int,
            commitScore: Int,
            cicdScore: Int,
        ): Score {
            return Score(
                analysisResult = analysisResult,
                readmeScore = readmeScore,
                testScore = testScore,
                commitScore = commitScore,
                cicdScore = cicdScore,
            )
        }
    }
}